src/plugins/intel_cpu/src/edge.cpp

```diff
@@ -292,7 +292,10 @@ void MKLDNNEdge::externalAllocate(MKLDNNWeightsSharing::Ptr weightsCache) {
     if (status != Status::NeedAllocation)
         return;
 
-    if (weightsCache) {
+    bool isTheOnlyChildEdge = getParent()->getChildEdges().size() == 1;
```

Comment:
Actually we need to check that there are peer edges on the same port. If the port of the edge is different the memory will not be shared.

Reply:
changed to
``` c++
getParent()->getChildEdgesAtPort(getInputNum()).size() == 1;
```